ast/src/analyzed/mod.rs

```diff
@@ -322,6 +322,30 @@ impl<T> Analyzed<T> {
             .filter_map(|(_poly, definition)| definition.as_mut())
             .for_each(|definition| definition.post_visit_expressions_mut(f))
     }
+
+    /// Retrieves (col_name, col_idx, offset) of each public witness in the trace.
+    pub fn get_publics(&self) -> Vec<(String, usize, usize)> {
+        let mut publics = self
+            .public_declarations
+            .values()
+            .map(|public_declaration| {
+                let witness_name = public_declaration.referenced_poly_name();
```

Comment:
This does not have to be a witness, public inputs can reference any column (even if it doesn't make sense in practice to expose a cell of a fixed column)

Reply:
probably `column` makes more sense in this block